account_payment_term_extension/models/account_payment_term.py

```diff
@@ -179,6 +179,10 @@ def compute(self, value, date_ref=False):
                 next_date += relativedelta(days=line.days,
                                            weeks=line.weeks,
                                            months=line.months)
+                if line.day_of_the_month > 0:
+                    months_delta = (line.day_of_the_month < next_date.day) and 1 or 0
+                    next_date += relativedelta(
+                        day=line.day_of_the_month, months=months_delta)
```

Comment:
```suggestion
                if line.day_of_the_month > 0:
                    months_delta = (line.months == 0) and (line.day_of_the_month < next_date.day) and 1 or 0
                    next_date += relativedelta(
                        day=line.day_of_the_month, months=months_delta)
```

On second thought, we should bypass the standard month_delta logic when `line.months` is set, since the delta for `line.months` is already taken care of in the previous step.